pandas/core/arrays/datetimes.py

```diff
@@ -83,6 +83,10 @@ def f(self):
         return result
 
     f.__name__ = name
+    if not docstring.startswith("\n"):
+        docstring = "".join(("\n", docstring))
+    if not docstring.endswith("\n"):
+        docstring = "".join((docstring, "\n"))
     f.__doc__ = docstring
```

Comment:
I'd make it as simple as:
```suggestion
f.__doc__ = '\n{}\n'.format(docstring)
```
Without the code you added.

Then, in the calls to `_field_accessor` we need to remove the `\n` and spaces around the description. I think that should work in all cases.

Reply:
oh, okay, i did think about this, then i thought if in the call to `_field_accessor`, we already used \n, then directly adding \n will be redundant... so i made a `if` judgement first.. 
but agree that if remove the `\n` added before, then this should not be the problem and look much simpler...

Reply:
we may have to do what you did if at some point we have something like:
```
_doc = """
Some summary.
"""
foo = _field_accessor(..., _doc)
```
But we'll complicate things at that point, if they ever happen. And to join two strings, I think it's better to use `'\n' + foo` of `'\n{}'.format(foo)` than using a join.

Reply:
okay, thanks for the tip!! @datapythonista